lib/auth/usertoken.go

```diff
@@ -510,3 +520,130 @@ func (s *Server) getRecoveryStartToken(ctx context.Context, tokenID string) (typ
 
 	return token, nil
 }
+
+// CreatePrivilegeToken returns a new privilege token after user successfully re-auth with their second factor.
+func (s *Server) CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (types.UserToken, error) {
+	authPref, err := s.GetAuthPreference(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if !authPref.GetAllowLocalAuth() {
+		return nil, trace.AccessDenied("local auth needs to be enabled")
+	}
+
+	secondFactor := authPref.GetSecondFactor()
+	if secondFactor == constants.SecondFactorOff {
+		return nil, trace.AccessDenied("second factor must be enabled")
+	}
+
+	// Begin authenticating user password or second factor.
+	switch req.GetSecondFactorCred().(type) {
+	case *proto.CreatePrivilegeTokenRequest_SecondFactorToken:
```

Comment:
Let's not duplicate MFA handling code; let's find a way to integrate this into the existing logic.